targets/psa_switch/externs/psa_meter.cpp

```diff
@@ -0,0 +1,52 @@
+/* Copyright 2020-present Cornell University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Yunhe Liu (yunheliu@cs.cornell.edu)
+ *
+ */
+
+#include "psa_meter.h"
+#include <iostream>
+
+namespace bm {
+
+namespace psa {
+
+void
+PSA_Meter::execute(const Data &index, Data &value) {
+    unsigned int color_out = _meter->execute_meter(get_packet(), index.get<size_t>(), (unsigned int) 0);
```

Comment:
please use `static_cast` instead of C-style casts. I'm surprised the style checker has not complained (maybe it's not enabled for this directory).

same below.